api-conversion/src/main/java/io/strimzi/kafka/api/conversion/converter/KafkaConverter.java

```diff
@@ -72,9 +72,9 @@ public ArrayOrObjectKafkaListeners apply(ArrayOrObjectKafkaListeners arrayOrObje
         Conversion.replaceLogging("/spec/entityOperator/topicOperator/logging", "log4j2.properties"),
         Conversion.replaceLogging("/spec/entityOperator/userOperator/logging", "log4j2.properties"),
         Conversion.replaceLogging("/spec/cruiseControl/logging", "log4j2.properties"),
-        new MetricsConversion<>("/spec/kafka", KafkaClusterSpec.class, "kafka"),
-        new MetricsConversion<>("/spec/zookeeper", ZookeeperClusterSpec.class, "zookeeper"),
-        new MetricsConversion<>("/spec/cruiseControl", CruiseControlSpec.class, "cruise-control")
+        new MetricsConversion<Kafka>("/spec/kafka", KafkaClusterSpec.class, "kafka"),
+        new MetricsConversion<Kafka>("/spec/zookeeper", ZookeeperClusterSpec.class, "zookeeper"),
+        new MetricsConversion<Kafka>("/spec/cruiseControl", CruiseControlSpec.class, "cruise-control")
```

Comment:
What was the problem here?

Reply:
Eclipse Java development tools flagged the assignment of `V1BETA1_TO_V1BETA2` with this error:
> `Type mismatch: cannot convert from VersionConversion<HasMetadata> to VersionConversion<Kafka>`

The added type parameter gives JDT an extra hint. I saw this error in both Eclipse itself as well as VSCode with the Java tools which I believe are based on JDT. I'd be curious to hear if others are using different IDEs like IntelliJ or older versions of Eclipse. This was seen with the current Eclipse release.